src/gen_rules.ml

```diff
@@ -28,30 +28,53 @@ module Gen(P : Params) = struct
      +-----------------------------------------------------------------+ *)
 
   module Eval_modules = Ordered_set_lang.Make(struct
-      type t = Module.t
-      let name = Module.name
+      type t = Loc.t * ((Module.t, string) result)
+      let name (_, m) =
+        match m with
+        | Error s -> s
+        | Ok m -> Module.name m
     end)
 
-  let parse_modules ~all_modules ~buildable =
+  let parse_modules ~(all_modules : Module.t String_map.t) ~buildable =
     let conf : Buildable.t = buildable in
-    let parse ~loc s =
+    let standard_modules =
+      String_map.map all_modules ~f:(fun m -> (Loc.none, Ok m)) in
+    let fake_modules = ref String_map.empty in
+    let parse : loc:Loc.t -> string -> (Loc.t * ((Module.t, string) result))
+      = fun ~loc s ->
         let s = String.capitalize_ascii s in
         match String_map.find s all_modules with
-        | Some m -> m
-        | None -> Loc.fail loc "Module %s doesn't exist." s
+        | Some m -> (loc, Ok m)
+        | None ->
+          if not (String_map.mem s !fake_modules) then (
+            fake_modules := String_map.add ~key:s ~data:loc !fake_modules;
```

Comment:
That will catch all modules right? Even the ones that are not excluded, such as in `(modules (fake))`. The error message will be wrong then?

Reply:
Ok, I understand. The included modules that are missing will be reported by `only_present_modules`.